app/controllers/api/apartments_controller.rb

```diff
@@ -0,0 +1,66 @@
+module Api
+  class ApartmentsController < Api::ApplicationController
+    def index
+      apartments = Apartment.all
+      render json: { apartments: }
+    end
+
+    def show
+      apartment = Apartment.find(params[:id])
+      render json: { apartment: }
+    end
+
+    def create
+      if current_user.role != 'admin'
+        render(json: { error: 'Only admin can create an apartment' }, status: 401)
```

Comment:
[SUGGESTION]: Please consider changing the error message to either "Only admin***s*** can create an apartment" or "Only ***an*** admin can create an apartment".